[2025-10-12T20:24:08.135+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: events_analytics_pipeline.extract_and_load_to_clickhouse scheduled__2025-10-12T19:00:00+00:00 [queued]>
[2025-10-12T20:24:08.138+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: events_analytics_pipeline.extract_and_load_to_clickhouse scheduled__2025-10-12T19:00:00+00:00 [queued]>
[2025-10-12T20:24:08.138+0000] {taskinstance.py:2170} INFO - Starting attempt 2 of 3
[2025-10-12T20:24:08.143+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): extract_and_load_to_clickhouse> on 2025-10-12 19:00:00+00:00
[2025-10-12T20:24:08.147+0000] {standard_task_runner.py:60} INFO - Started process 2141 to run task
[2025-10-12T20:24:08.149+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'events_analytics_pipeline', 'extract_and_load_to_clickhouse', 'scheduled__2025-10-12T19:00:00+00:00', '--job-id', '22', '--raw', '--subdir', 'DAGS_FOLDER/events_pipeline_dag.py', '--cfg-path', '/tmp/tmp_182hhxj']
[2025-10-12T20:24:08.151+0000] {standard_task_runner.py:88} INFO - Job 22: Subtask extract_and_load_to_clickhouse
[2025-10-12T20:24:08.179+0000] {task_command.py:423} INFO - Running <TaskInstance: events_analytics_pipeline.extract_and_load_to_clickhouse scheduled__2025-10-12T19:00:00+00:00 [running]> on host 33426d0e8621
[2025-10-12T20:24:08.212+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-team' AIRFLOW_CTX_DAG_ID='events_analytics_pipeline' AIRFLOW_CTX_TASK_ID='extract_and_load_to_clickhouse' AIRFLOW_CTX_EXECUTION_DATE='2025-10-12T19:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-12T19:00:00+00:00'
[2025-10-12T20:24:08.213+0000] {events_pipeline_dag.py:80} INFO - Connecting to PostgreSQL...
[2025-10-12T20:24:08.214+0000] {events_pipeline_dag.py:84} INFO - Extracting data from PostgreSQL...
[2025-10-12T20:24:08.464+0000] {events_pipeline_dag.py:92} INFO - Extracted 21300 rows from PostgreSQL
[2025-10-12T20:24:08.464+0000] {events_pipeline_dag.py:95} INFO - Connecting to ClickHouse...
[2025-10-12T20:24:08.464+0000] {events_pipeline_dag.py:99} INFO - Clearing existing data...
[2025-10-12T20:24:08.467+0000] {events_pipeline_dag.py:103} INFO - Loading data into ClickHouse...
[2025-10-12T20:24:08.477+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/events_pipeline_dag.py", line 106, in extract_and_load_to_clickhouse
    client.execute(
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/client.py", line 376, in execute
    rv = self.process_insert_query(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/client.py", line 607, in process_insert_query
    rv = self.send_data(sample_block, data,
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/client.py", line 660, in send_data
    self.connection.send_data(block)
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/connection.py", line 688, in send_data
    self.block_out.write(block)
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/streams/native.py", line 48, in write
    write_column(self.context, col_name, col_type, items,
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/columns/service.py", line 167, in write_column
    column.write_data(items, buf)
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/columns/base.py", line 145, in write_data
    self._write_data(items, buf)
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/columns/base.py", line 149, in _write_data
    self.write_items(prepared, buf)
  File "/home/airflow/.local/lib/python3.11/site-packages/clickhouse_driver/columns/stringcolumn.py", line 18, in write_items
    buf.write_strings(items, encoding=self.encoding)
  File "clickhouse_driver/bufferedwriter.pyx", line 54, in clickhouse_driver.bufferedwriter.BufferedWriter.write_strings
AttributeError: 'UUID' object has no attribute 'encode'
[2025-10-12T20:24:08.482+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=events_analytics_pipeline, task_id=extract_and_load_to_clickhouse, execution_date=20251012T190000, start_date=20251012T202408, end_date=20251012T202408
[2025-10-12T20:24:08.487+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 22 for task extract_and_load_to_clickhouse ('UUID' object has no attribute 'encode'; 2141)
[2025-10-12T20:24:08.496+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-10-12T20:24:08.505+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
